text/profiles/0237-profiles-data-model.md

```diff
@@ -0,0 +1,1067 @@
+# Profiles Data Format
+
+Introduces Data Model for Profiles signal to OpenTelemetry.
+
+<!-- toc -->
+* [Motivation](#motivation)
+* [Design Notes](#design-notes)
+  * [Design Goals](#design-goals)
+* [Data Model](#data-model)
+  * [Relationships Diagram](#relationships-diagram)
+  * [Relationships With Other Signals](#relationships-with-other-signals)
+    * [From profiles to other signals](#from-profiles-to-other-signals)
+    * [From other signals to profiles](#from-other-signals-to-profiles)
+  * [Proto Definition](#proto-definition)
+  * [Message Descriptions](#message-descriptions)
+    * [Message `ProfilesData`](#message-profilesdata)
+    * [Message `ResourceProfiles`](#message-resourceprofiles)
+    * [Message `ScopeProfiles`](#message-scopeprofiles)
+    * [Message `Profile`](#message-profile)
+    * [Message `ProfileType`](#message-profiletype)
+    * [`Sample` structure](#sample-structure)
+    * [Message `Stacktrace`](#message-stacktrace)
+    * [Message `AttributeSet`](#message-attributeset)
+    * [Message `Link`](#message-link)
+    * [Message `Location`](#message-location)
+    * [Message `Mapping`](#message-mapping)
+    * [Message `Function`](#message-function)
+  * [Example Payloads](#example-payloads)
+    * [Simple example](#simple-example)
+  * [Notable differences compared to other signals](#notable-differences-compared-to-other-signals)
+    * [Relationships between messages](#relationships-between-messages)
+    * [Arrays of Integers vs Arrays of Structures](#arrays-of-integers-vs-arrays-of-structures)
+* [Trade-offs and mitigations](#trade-offs-and-mitigations)
+* [Prior art and alternatives](#prior-art-and-alternatives)
+  * [pprof](#pprof)
```

Comment:
I am curious, why is pprof the only format mentioned in prior art? Where there considerations given to other formats, e.g.
  * Chromium's Trace Event Format
  * Common Trace Format
  * Linux perf.data
  * etc

Most of these can be directly consumed by popular tools like Perfetto, chrome://

Reply:
> I am curious, why is pprof the only format mentioned in prior art? Where there considerations given to other formats, e.g.

Yeah, maybe a few more formats should be mentioned in this OTEP.

But there is a rather large number of [profiling formats](https://profilerpedia.markhansen.co.nz/formats/), so it's a bit difficult to make a selection. The profiling WG is definitely aware of the formats you mentioned. A few quick thoughts on them from my end:

> * Chromium's Trace Event Format

JSON based, not competitive on payload size, no support for aggregation, weak spec (not machine readable, different implementations e.g. perfetto/catapult end up interpreting it differently).

> * Common Trace Format

More of a meta system for creating tracing formats. Very interesting, but extremely complex.

> * Linux perf.data

Probably too low-level ("On a high level perf.data contains the events generated by the PMUs, plus metadata"). Many of the fields only make sense for data collected by a kernel.

Reply:
> > * Linux perf.data
> 
> Probably too low-level ("On a high level perf.data contains the events generated by the PMUs, plus metadata"). Many of the fields only make sense for data collected by a kernel.

FWIW I also find perf.data format incomprehensible (or non-intuitive at least) but this could be just me.

Reply:
> Yeah, maybe a few more formats should be mentioned in this OTEP.

Yes, I think even these very brief evaluations give more credibility to the proposal indicating that other formats were looked at and not selected for specific reasons.